tests/store/artifact/test_cloud_artifact_repo.py

```diff
@@ -16,3 +23,67 @@ def test_chunk_size_validation_failure():
         _validate_chunk_size_aws(5 * 1024**2 - 1)
     with pytest.raises(MlflowException, match="Multipart chunk size"):
         _validate_chunk_size_aws(5 * 1024**3 + 1)
+
+@pytest.mark.parametrize("future_result, expected_call_count", [
+    (None, 2),  # Simulate where creds are expired, but successfully download after refresh
+    (Exception("fake_exception"), _MLFLOW_MPD_NUM_RETRIES.get() + 1)  # Simulate where there is a download failure and retries are exhausted
```

Comment:
Do not use the environment variable here, set an arbitrary number and use `monkeypatch.setenv` to set the variable to suit your needs.

Reply:
done